internal/appconfig/validation.go

```diff
@@ -64,3 +73,30 @@ func (cfg *Config) Validate(ctx context.Context) (err error, extra_info string)
 	}
 
 }
+
+func (cfg *Config) BuildStrategies() []string {
+	strategies := []string{}
+
+	if cfg.Build == nil {
```

Comment:
This is not tested. 

A common trap here is to forget to check for nil cfg, like in `cfg == nil || cfg.Build == nil`

Reply:
interesting, TIL